js/main.js

```diff
@@ -75,12 +102,40 @@ function viewSwap(view) {
     ent.classList.add('hidden');
   }
 }
+
 const a = document.querySelector('a');
 a.addEventListener('click', function () {
   viewSwap('entries');
 });
 
 const newEntryBtn = document.querySelector('.new');
 newEntryBtn.addEventListener('click', function () {
+  resetForm();
+  viewSwap('entry-form');
+
+});
+
+ul.addEventListener('click', function (event) {
+  if (!event.target.matches('.fa-solid')) {
+    return;
+  }
   viewSwap('entry-form');
+  for (let i = 0; i < data.entries.length; i++) {
+    if (data.entries[i].entryId === Number(event.target.closest('li').getAttribute('data-entry-id'))) {
+      data.editing = data.entries[i];
+      const prePopulate = {
+        $t: data.editing.$title,
+        $u: data.editing.$url,
+        $n: data.editing.$notes
+      };
+      const getTitle = document.querySelector('#title');
+      getTitle.setAttribute('value', prePopulate.$t);
+      const getPhoto = document.querySelector('#url');
+      getPhoto.setAttribute('value', prePopulate.$u);
+      previewPhoto.setAttribute('src', prePopulate.$u);
+      const getNote = document.querySelector('#notes');
+      getNote.innerHTML = prePopulate.$n;
```

Comment:
instead of using the `setAttribute` calls here, use the same pattern as on line 28... so for example:

`form.elements.title.value = prePopulate.$t;`

do the same for the `notes` and `url` field

line 135 is fine the way it is.

Fixing this will fix the issue we ran into with the form data persisting even though it should be reset when clicking the "new" button. Using the `setAttribute` method will prevent the `form.reset()` function call to reset back to the default value, but will instead use the last set attribute from the `setAttribute` call. See this [link](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reset) in the 3rd paragraph.